renderite/serialization.zig

```diff
@@ -32,6 +32,17 @@ pub const IpcDeserializer = struct {
         }
     }
 
+    pub fn readList(self: IpcDeserializer, comptime T: type) !T {
+        const base_t = switch (@typeInfo(T)) {
+            .array => |array| array.child,
+            .pointer => |pointer| pointer.child,
+            // else => @compileError(std.fmt.comptimePrint("Unsupported type {s} is not an array (was {s})", .{ @typeName(T), @tagName(@typeInfo(T)) })),
+            else => return undefined, // TODO
+        };
+
+        return try self.readValueList(base_t, self.gpa);
```

Comment:
Types are `PascalCase`

```suggestion
        const BaseType = switch (@typeInfo(T)) {
            .array => |array| array.child,
            .pointer => |pointer| pointer.child,
            // else => @compileError(std.fmt.comptimePrint("Unsupported type {s} is not an array (was {s})", .{ @typeName(T), @tagName(@typeInfo(T)) })),
            else => return undefined, // TODO
        };

        return try self.readValueList(BaseType, self.gpa);
```